java/google/registry/rde/imports/RdeHostLinkAction.java

```diff
@@ -117,10 +117,18 @@ public void map(JaxbFragment<XjcRdeHostElement> fragment) {
         ofy().transact(new VoidWork() {
           @Override
           public void vrun() {
+            // link host to superordinate domain and set time of last superordinate change to
+            // the time of the import
             HostResource host =
                 ofy().load().now(Key.create(HostResource.class, xjcHost.getRoid()));
             ofy().save()
-                .entity(host.asBuilder().setSuperordinateDomain(superordinateDomainKey).build());
+                .entity(host.asBuilder().setSuperordinateDomain(superordinateDomainKey)
+                    .setLastSuperordinateChange(ofy().getTransactionTime())
+                    .build());
+            // link domain to subordinate host
+            DomainResource domain = ofy().load().now(superordinateDomainKey);
```

Comment:
You need to grab the superordinate domain key off the host that you just loaded inside this transaction, in case anything else has written out a change to the host since you retrieved it.

It may be simplest to perform all actions inside a single transaction.